torchmdnet/module.py

```diff
@@ -57,34 +57,95 @@ def configure_optimizers(self):
         }
         return [optimizer], [lr_scheduler]
 
-    def forward(self,
-                z: Tensor,
-                pos: Tensor,
-                batch: Optional[Tensor] = None,
-                q: Optional[Tensor] = None,
-                s: Optional[Tensor] = None,
-                extra_args: Optional[Dict[str, Tensor]] = None
-                ) -> Tuple[Tensor, Optional[Tensor]]:
-
+    def forward(
+        self,
+        z: Tensor,
+        pos: Tensor,
+        batch: Optional[Tensor] = None,
+        q: Optional[Tensor] = None,
+        s: Optional[Tensor] = None,
+        extra_args: Optional[Dict[str, Tensor]] = None,
+    ) -> Tuple[Tensor, Optional[Tensor]]:
         return self.model(z, pos, batch=batch, q=q, s=s, extra_args=extra_args)
 
     def training_step(self, batch, batch_idx):
-        return self.step(batch, mse_loss, "train")
+        return self.step(batch, [mse_loss], "train")
 
     def validation_step(self, batch, batch_idx, *args):
-        if len(args) == 0 or (len(args) > 0 and args[0] == 0):
-            # validation step
-            return self.step(batch, mse_loss, "val")
-        # test step
-        return self.step(batch, l1_loss, "test")
+        # If args is not empty the first (and only) element is the dataloader_idx
+        # We want to test every number of epochs just for reporting, but this is not supported by Lightning.
+        # Instead, we trick it by providing two validation dataloaders and interpreting the second one as test.
+        # The dataloader takes care of sending the two sets only when the second one is needed.
+        is_val = len(args) == 0 or (len(args) > 0 and args[0] == 0)
+        step_type = (
```

Comment:
can you write this as a normal if/else please? It's less lines and more readable than a comprehension